### PR TITLE
Add integration test for turtlebot

### LINKED ISSUE #171: Add section in README referencing isar-turtlebot

### PR DESCRIPTION
**Pull request by aestene**
Closes https://github.com/equinor/isar-turtlebot/issues/38
Closes #171

### CODE REVIEW

**Comment by aestene** [comment]
The current state of the test requires it to run locally with a simulator already running. It does not run as part of the pipeline yet and is excluded when running `pytest`

File: README.md [hunk 1]
```diff
@@ -164,18 +179,34 @@ AZURE_CLIENT_SECRET
...
+
+To create an interface test in your robot repository, use the function `interface_test` from `robot_interface`. The
+argument should be an interface object from your robot specific implementation.
+See [isar-robot](https://github.com/equinor/isar-robot/blob/main/tests/interfaces/test_robotinterface.py) for example.
+
+### Integration tests
+
+Integration tests can be found [here](https://github.com/equinor/isar/tree/main/tests/integration) and have been created
+with a simulator in mind. The integration tests will not run as part of `pytest .` or as part of the CI/CD pipeline. To
+run the integration tests please follow the instructions in blabla for setting up the `isar-turtlebot` implementation
```
Comment (tsundvoll) [inline README.md:193]: Maybe fill in for 'blabla' here? :)

File: tests/integration/turtlebot/test_successful_mission.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    return StateMachineThread(injector=injector_turtlebot)
+
+
+@pytest.fixture(autouse=True)
+def run_before_and_after_tests():
+    results_folder: Path = Path("tests/results")
+    yield
+
+    print("Removing temporary results folder for testing")
+    shutil.rmtree(results_folder)
```
Comment (tsundvoll) [inline tests/integration/turtlebot/test_successful_mission.py:80]: This gives me a `FileNotFoundError` when 'tests/results' folder does not exist. I had not the simulator running or the isar-turtlebot installed yet though. But still, maybe it should handle the case where the folder does not exist?

File: tests/integration/turtlebot/test_successful_mission.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        f"schedule/start-mission?ID={_id}",
+        headers={"Authorization": "Bearer {}".format(access_token)},
+    )
+    time.sleep(5)
+
+    mission_id: str = state_machine_thread.state_machine.current_mission.id
+    mission: Mission = deepcopy(state_machine_thread.state_machine.current_mission)
+
+    start_time: datetime = datetime.utcnow()
+    while state_machine_thread.state_machine.current_state != "idle":
```
Comment (sondreo) [inline tests/integration/turtlebot/test_successful_mission.py:103]:
```suggestion
    while state_machine_thread.state_machine.current_state != States.Idle:
```

File: tests/integration/turtlebot/test_successful_mission.py [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+    LocalStorageModule,
+    QueuesModule,
+    ReaderModule,
+    RequestHandlerModule,
+    RobotModule,
+    ServiceModule,
+    StateMachineModule,
+    UtilitiesModule,
+)
+from isar.services.readers.base_reader import BaseReader
```
Comment (sondreo) [inline tests/integration/turtlebot/test_successful_mission.py:30]:
```suggestion
from isar.services.readers.base_reader import BaseReader
from isar.state_machine.states_enum import States
```

**Review by tsundvoll — COMMENTED** [review]
The integration test works very well here when I first had set up the simulator to run beside :)

**Review by sondreo — CHANGES_REQUESTED** [review]
Looks good! The integration test passes when using a local instance of the simulator already running.

See some minor suggestions.

File: README.md (hunk 1 above)
Comment (aestene) [reply README.md:193]: Oh well, there was supposed to be a link to the other section there once I had made it :D

File: tests/integration/turtlebot/test_successful_mission.py (hunk 2 above)
Comment (aestene) [reply tests/integration/turtlebot/test_successful_mission.py]: Done.

File: tests/integration/turtlebot/test_successful_mission.py (hunk 3 above)
Comment (aestene) [reply tests/integration/turtlebot/test_successful_mission.py:104]: Done

**Review by sondreo — APPROVED** [review]
LGTM

**Pull request merged**